sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_client_connection.py

```diff
@@ -2111,15 +2072,15 @@ def Batch(
         if is_error:
             raise exceptions.CosmosBatchOperationError(
                 error_index=error_index,
-                headers=self.last_response_headers,
+                headers=last_response_headers,
                 status_code=error_status,
                 message="There was an error in the transactional batch on index {}. Error message: {}".format(
                     str(error_index),
                     Constants.ERROR_TRANSLATIONS.get(error_status)
                 ),
                 operation_responses=final_responses
             )
-        return final_responses
+        return cast(List[Dict[str, Any]], final_responses)
```

Comment:
Interesting that we needed to cast both here and in the container client? I would've thought only once would be necessary...